Fix issues with prefix searches

Niklas-Deworetzki opened the pull request:
Small set of fixes for prefix search functionality. Fixes #16.

The following changes are included:
- A small test suite containing integration tests for corpus search functionality
- Prefix queries matching more than 10% of the corpus are no longer skipped
- Prefix queries with multiple tokens no longer return the wrong duplicated features (see #16)

heatherleaf commented:
Nice!
Some comments below

heatherleaf commented:
What's your suggested way of running the tests? Just running this doesn't do anything:
```
python -m tests.test_basic_search
```

The `unittest` documentation suggests to have this at the end of a file:
```
if __name__ == '__main__':
    unittest.main()
```

heatherleaf commented:
You use `NamedTemporaryFile` with the argument `delete_on_close`. This was added in Python 3.12, so the question is if we want to enforce that. I'm not certain; 3.11 is still quite recent. Unless there are some features of 3.12 that are vital.

heatherleaf commented:
Maybe related to delete_on_close. I get the following warning (3 times when running test_basic_search):
```
/Users/peb/Repositories/korpsearch/corpus_reader.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='/var/folders/jf/m2x9llfx56xccz26rz6ydxd00000gn/T/tmpg2w_jdvv.csv'>
```

heatherleaf reviewed (commented):
See my comments about the testing, the rest looks fine

Niklas-Deworetzki commented:
Oh, I didn't check which Python versions are required for different features. I'll make sure to not use any features that require the newest versions. And I will insert the call to `unittest.main`, of course.

Which is the highest Python version we plan to support? Maybe we can do automatic checks to enforce support for that version.

heatherleaf commented:
> Which is the highest Python version we plan to support? Maybe we can do automatic checks to enforce support for that version.

Perhaps 3.11? 
It's anyway a version most people *should* have since it's ~30-50% faster than 3.10

Niklas-Deworetzki commented:
I addressed your comments. The calls to `unittest.main` are now present and the code is compatible with Python 3.11

The ResourceWarnings are unrelated to the changes, but I think I found the reason for them as well. I will create an issue in the afternoon and provide a patch for that as well.
